src/main/java/com/sap/cloudfoundry/client/facade/util/UriUtils.java

```diff
@@ -0,0 +1,16 @@
+package com.sap.cloudfoundry.client.facade.util;
+
+import java.nio.charset.Charset;
+import java.util.List;
+
+public class UriUtils {
+
+    public static String encodeChars(String queryParam, List<String> charsToEncode) {
+        return charsToEncode.stream()
+                            .reduce(queryParam, UriUtils::encodeChar);
```

Comment:
The reduction operation isn't intended for what we do here. Although fancy looking, this is sort of a hack.
My recommendation is to do like so:
```java
for (String charToEncode : charsToEncode) {
   queryParam = queryParam.replaceAll(charToEncode, spring.UriUtils.encode(charToEncode));
}
return queryParam;
```

Reply:
+1